src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py

```diff
@@ -57,31 +53,19 @@ def to_sql(value: Any, datatype: DataType, from_values_statement: bool = False)
             return "NULL"
     if isinstance(datatype, BinaryType):
         if value is None:
-            return "NULL :: binary"
-    if isinstance(datatype, IntegerType):
+            return "NULL :: BINARY"
```

Comment:
Changes here are nor really related to literal value - it's just to make translated sql more formal (https://docs.snowflake.com/en/sql-reference/data-types-numeric.html#int-integer-bigint-smallint-tinyint-byteint, we can just use INT and FLOAT) and readable (uppercase).